examples/llama.py

```diff
@@ -181,6 +185,26 @@ def sample(logits, temperature):
     probs = probs.numpy().flatten()
     return int(np.random.choice(len(probs), p=probs))
 
+def concat_weights(models):
+  def convert(name) -> Tensor:
+    disk_tensors = [model[name] for model in models]
+    if len(disk_tensors) == 1:
+      return disk_tensors[0]
+    if len(disk_tensors[0].shape) == 1:
+      return disk_tensors[0]
+    if name.startswith('tok_embeddings.') or name.endswith('.attention.wo.weight') or name.endswith('.feed_forward.w2.weight'):
+      axis = 1
+    else:
+      axis = 0
+    lazy_tensors = [data.to(device=Device.DEFAULT) for data in disk_tensors]
+    first, rest = lazy_tensors[0], lazy_tensors[1:]
+    return first.cat(*rest, dim=axis).realize()
+  ret = {}
+  for name in (t := tqdm({name: None for model in models for name in model})):
```

Comment:
Won't this try to load the same weight multiple times?

Reply:
Hahaha I just caught that myself too.  It doesn't load twice, but it's extraneous and not needed (addressed with the latest commit).